crates/c/src/lib.rs

```diff
@@ -1031,6 +1246,35 @@ impl<'a> wit_bindgen_core::InterfaceGenerator<'a> for InterfaceGenerator<'a> {
         self.finish_ty(id, prev);
     }
 
+    fn type_resource(&mut self, id: TypeId, name: &str, docs: &Docs) {
+        let prev = mem::take(&mut self.src.h_defs);
+
+        if !self.in_import {
```

Comment:
@alexcrichton I believe I've mostly addressed this and the related concern below.  However, there's still a case where we will generate a `struct` for an alias instead of just a `typedef` alias: when there's no type to alias.  Specifically, if you have a `resource a` and a `type b = a`, and then `foo: func(b: borrow<b>)`, but `borrow<a>` never appears anywhere, then `Resolve::types` will not contain a `TypeDefKind::Handle` for `a`, so we won't generate a type for it, and thus there's nothing to alias.  In that case, we just generate a `struct` for `b`.

In other words, we're not really aliasing resource types -- we're aliasing handle types.  And that means we can only generate a `typedef` alias for a handle type if that handle type actually exists.  Technically we could synthesize a type, but that would be a pretty big lift -- we can't really add new types to `Resolve` the way the code is currently structured, and yet everything takes a `TypeId`.

I'm open to alternative approaches, but I feel like what I've described above works well enough for the time being.

Reply:
I think that's fine yeah, I'd like to dig in more but I'm happy to do that as a follow-up!